src/agent/agent_loop.py

```
@@ -0,0 +1,340 @@
+                    tc_json = json.dumps(assistant_msg.get("tool_calls", []), ensure_ascii=False)
+                    db.save_message(
+                        current_session_id, "assistant",
+                        assistant_msg.get("content") or "",
+                        tool_calls_json=tc_json
+                    )
+
+                    for tool_call in message.tool_calls:
+                        func_name = tool_call.function.name
+                        try:
+                            args = json.loads(tool_call.function.arguments)
+                        except:
```

Ruff (0.16.1)

[error] Do not use bare `except`

(E722)